### PR TITLE
[dunfell] cherry pick snapd patches from kirkstone/master, update to 2.57.5

### PR DESCRIPTION
**Pull request by bboozzoo**
Cherry pick relevant patches from kirkstone branch. This includes the refactor to split the recipes into smaller, more manageable chunks and an update 2.57.5. The kirkstone patches that touched kernel recipes were skipped. At this point the snapd recipes in dunfell/kirkstone/master are in sync.

### CODE REVIEW

File: recipes-support/snapd/snapd-go.inc [hunk 1]
```diff
@@ -0,0 +1,72 @@
+inherit go-mod
+
```
Comment (zyga) [inline recipes-support/snapd/snapd-go.inc:2]: Do we need to put `do_compile[network] = "1"` anywhere?

**Review by zyga — COMMENTED** [review]
One comment about the (upcoming) prompter program/services.

File: recipes-support/snapd/snapd.inc [hunk 2]
```diff
@@ -138,13 +99,13 @@ do_install() {
 	rm -fv ${D}${systemd_unitdir}/system/snapd.system-shutdown.service
 	rm -fv ${D}${systemd_unitdir}/system/snapd.snap-repair.*
 	rm -fv ${D}${systemd_unitdir}/system/snapd.core-fixup.*
-	rm -fv ${D}${systemd_unitdir}/snapd.recovery-chooser-trigger.service
+	rm -fv ${D}${systemd_unitdir}/system/snapd.recovery-chooser-trigger.service
+	rm -fv ${D}${systemd_unitdir}/system/snapd.aa-prompt-listener.service
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply recipes-support/snapd/snapd.inc:103]: we're not installing snapd-aa-prompt-listener at all

File: recipes-support/snapd/snapd-go.inc (hunk 1 above)
Comment (bboozzoo) [reply recipes-support/snapd/snapd-go.inc:2]: no, the recipe uses snapd with vendored dependencies, so everything is included already in the tarball

**Pull request merged**